File: padroes-de-projetos/factory-singleton/singleton.py
# Singleton
class Incremental:
	contador = 0
	numero = contador

	@staticmethod
	def incrementar():
		Incremental.contador = Incremental.contador + 1


# Singleton Pessoa
class Pessoa(object):
	_instancia = None

	def __init__(self, nome: str) -> None:
		self.nome = nome

	@staticmethod
	def getInstancia():
		if Pessoa._instancia is None:
			Pessoa._instancia = Pessoa('Joffily')

		return Pessoa._instancia


if __name__ == "__main__":
	# Incrementa uma vez
	Incremental().incrementar()
	print(Incremental.contador)

	#Incrementa a segunda vez
	Incremental().incrementar()
	print(Incremental.contador)


	pessoa = Pessoa.getInstancia()
	pessoa2 = Pessoa.getInstancia()

	print(pessoa.nome)
	# Mostramos os IDs dos objetos
	print(id(pessoa))
	print(id(pessoa2))

File: padroes-de-projetos/factory-singleton/Singleton.java
/*
* @author Marianna V, Joffily F
* @program Singleton exercice
*/

class Incremental {
	private static int contador = 0;
	private int numero;
	private static Incremental instancia;

	private Incremental() {
	}

	public static Incremental getInstancia() {
		if (instancia == null) {
			instancia = new Incremental();
		}

		return instancia;
	}

	public void mostra() {
		System.out.println(++contador);
	}
}

public class Singleton {

	public static void main(String[] args) {

		for (int i = 0; i < 5; i++) {
			Incremental inc = Incremental.getInstancia();
			inc.mostra();
		}
	}
}
